pkg/catalog/catalog.go

```diff
@@ -261,6 +261,18 @@ func (c *Catalog) CreateBareRepository(ctx context.Context, repository string, s
 	return catalogRepo, nil
 }
 
+func (c *Catalog) getRepository(ctx context.Context, repository string) (*graveler.RepositoryRecord, error) {
```

Comment:
Prefer to use the errors from graveler if possible and remove the one in the catalog level.
If we keep the catalog level - we need to covert also branch/tag/commit - so in that case, something like the follow that will use switch case to convert graveler errors to catalog.
```go
func convertGravelerError(err error) error
```